### DISCUSSION TITLE
MHP30 repairable if not heating?

### DISCUSSION THREAD

**Discussion by jrowberg** [Ralim/IronOS#1340] [Q&A]
Hi all,

I'm reaching a bit here, but this dev crew seems like the most knowledgeable (and willing to share that knowledge). I got a Miniware MHP30 back in February, and I've used it many times with great satisfaction. I tried again earlier this week, and suddenly it won't heat up at all anymore. Everything appears to function correctly, except when it's supposed to get warmer, it doesn't. It just runs in "heating" mode forever.

I tried updating the firmware with the latest official 2.06 release, and then I discovered this project and tried the latest compatible IronOS release, but the same thing happened: it flashed fine, and appears to work perfectly, but...no heat. The device is still inside the 1-year warranty, and I've actually got good, quick support from the manufacturer, and they're going to replace the base for me.

But I would bet there's probably one simple failed component here. I figure if you've written custom firmware for this exact hardware, someone here probably knows what the schematic looks like (roughly anyway) and whether there's a main culprit. I'd love to repair it if possible, both to avoid wasting otherwise good electronics and to have something up and running possibly faster than a replacement will arrive.

Any ideas?

Thanks for any help you can provide!

**Comment by TimoteoMendes** [comment]
Hello,

Did you had any luck?
I am looking for a schematic of the unit as the display simbol appear and nothing else is after.
Was searching the web but no lucky for now, by any chance do you have that document?
Thanks

**Reply by Ralim** [reply]
No schematics that I know of sadly.
What do you get on screen? Does it enter DFU mode?

**Reply by TimoteoMendes** [reply]
Hello Ralim,
Thanks for the reply. 
It is very sad that could not found any schematics, was expecting as the reverse engineer was done that someone had it.
While was testing the board, when power on, the miniware simbol appear for a couple of seconds then all black, that makes me thinking it was an firmware issue.
Try to upload using DFU mode but always had the .ERR so connect an st-link and upload the .hex to upgrade. As nothing changes try to clean the chip and with it, another problem start as it was not possible to start it at all, by chance to do know how I can reverse it as I do not have done the backup?
After some more search I found your DFU file that could be upload and using the st-link once again upload the dfu file, but this part I didn't understand as it does not enter in DFU... but at least it if running because the green light is on and if I presse the A button it start to heat but without LCD. Try to check the I2C communications with st-link but the o-led is not answering, maybe it is really defective.
Do you know where I can found a replacement part for this?
Thanks,
Timóteo

**Reply by Ralim** [reply]
So if you didn't take a firmware flash backup, you will have wiped Miniware's DFU bootloader.
Your options would be to get someone elses backup and patch out the CPU-ID check OR you can try using your stlink to load this alternative bootloader instead https://github.com/Ralim/IronOS-dfu.
If you get that to load, when you enter programming mode it will show up as a USB-DFU device (NOT a virtual storage disk). You can then use dfu-util to flash IronOS hex files or IronOS dfu files.
I dont think the stock firmware will work as you will be missing the magic bytes for the CPU-ID check as these are stored at the very end of the bootloader area.

Getting the light to come on may not mean code is running correctly, as spurrious signals on the pin easily get random colours with the WS2812 type led's.


If you were getting the miniware logo then it seems quite possible the OLED itself is _ok_.
If it is damaged, Aliexpress has compatible ones afaik.

**Reply by TimoteoMendes** [reply]
By chance, do you have the file to share with me? And can you explain how and where to change the ID.
Your's dfu is the one the is now loaded on the MHP30 with the latest EN revision.

About the LED it looks ok as it start green and after press the A it changes to red fad in and get's hot, if press B for a long time it turns yellow (that is hot).

The log was the first indication of a good OLED signal, but with this firmware I get nothing at all.
Do you have a reference I can look for in Aliexpress, have try MHP30 OLED without luck. Also have used OLED I2C but didn't hapear any like the original.

Thanks
